typescript/web/src/components/dataset-classes/dataset-classes.context.tsx

```diff
@@ -28,6 +28,8 @@ export interface DatasetClassesState {
   loading: boolean;
   onReorder: OnReorderCallback;
   onExportClasses(): void;
+  setHasEditedClassName(value: boolean): void;
+  hasEditedClassName: boolean;
```

Comment:
We don't need to have a state (and trigger extra renders) for this.
Instead, we should test whether the input value is equal to the `editClass?.name`.